packages/core/content-type-builder/admin/src/components/FormModalSubHeader/index.js

```diff
@@ -0,0 +1,74 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import { useIntl } from 'react-intl';
+import { Typography } from '@strapi/design-system/Typography';
+import upperFirst from 'lodash/upperFirst';
+import { getTrad } from '../../utils';
+import getModalTitleSubHeader from './getModalTitleSubHeader';
+
+const FormModalSubHeader = ({
+  actionType,
+  modalType,
+  forTarget,
+  kind,
+  step,
+  attributeType,
+  attributeName,
+  customField,
+}) => {
+  const { formatMessage } = useIntl();
+  const type =
+    modalType === 'customField'
+      ? upperFirst(formatMessage(customField.intlLabel))
+      : upperFirst(
+          formatMessage({
+            id: getTrad(`attribute.${attributeType}`),
+          })
+        );
+
+  return (
+    <Typography as="h2" variant="beta">
+      {formatMessage(
+        {
+          id: getModalTitleSubHeader({
+            actionType,
+            forTarget,
+            kind,
+            step,
+            modalType,
+          }),
+          defaultMessage: 'Add new field',
+        },
+        {
+          type,
+          name: upperFirst(attributeName),
+          step,
```

Comment:
Maybe I am missing something here, but what do you need `type`, `name` and `step` for?

Reply:
They are needed depending on the returned value from `getModalTitleSubheader`. For customFields we need `type` and `name`.`type` is used when the `actionType` is create and `name` is used when the `actionType` is edit